clang/lib/Driver/Driver.cpp

```diff
@@ -5515,6 +5579,14 @@ class OffloadingActionBuilder final {
       delete SB;
   }
 
+  void pushForeignAction(Action *A) {
```

Comment:
Could you add a comment describing what a 'foreign action' is and how it is handled (as opposed to native files)

Reply:
Hi @mdtoguchi,
thanks for your review. I added a [commit](https://github.com/intel/llvm/pull/6395/commits/cdf91e40c206c541f6dc4d250b5e0ebf7c6033bd) with some comments, let me know how they sound.

The foreign actions are actions coming from different specialized device action builder.
Basically, we passed actions form the `CudaActionBuilder` to the `SYCLActionBuilder` making them communicating together in order to compile CUDA sources with SYCL.